backend/battling/views.py

```diff
@@ -22,22 +23,26 @@ class CreateBattleView(LoginRequiredMixin, CreateView):
     form_class = CreateBattleForm
     template_name = "battling/create_battle.html"
 
+    def get_initial(self):
+        obj_creator = self.request.user
+        self.initial = {"creator": obj_creator}
+        return self.initial
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        users = User.objects.all()
+        context["users"] = users
+        return context
+
     def form_valid(self, form):
         form.instance.creator = self.request.user
         form.instance.status = Battle.BattleStatus.ONGOING
 
         battle = form.save()
 
-        team_creator = Team.objects.create(battle=battle, trainer=self.request.user)
-
-        team_opponent = Team.objects.create(battle=battle, trainer=battle.opponent)
-
-        send_battle_invite(battle, team_opponent.id)
+        team_creator_id = Team.objects.get(battle=battle, trainer=self.request.user).id
```

Comment:
I think this is unnecessary. You have already it in `battle.creator.id`

Reply:
battle.creator.id gives me de creator id, team_creator_id gives me de id of the creator's team

Reply:
The variable name got me here. It should be `creator_team_id`
OK, code is right then, but it is a bit wasteful. If all you need is id, you can do `team_creator_id = Team.objects.get(battle=battle, trainer=self.request.user).only('id').id` (otherwise it is  `SELECT <all fields>` in SQL)

Reply:
Yeah, I just realized that and fixed the variable name... but gonna change to what you suggested now